### ISSUE TITLE
Flaky Test `DeploymentClusteredTest.shouldRedistributeDeploymentWhenDeploymentPartitionIsRestarted`

### ISSUE THREAD

**Issue by mustafadagher** [camunda/camunda#17303]
**Summary**

- How often does the test fail? [once](https://github.com/camunda/zeebe/actions/runs/8552447128/attempts/1#summary-23433567011)
- Does it block your work? no
- Do we suspect that it is a real failure? IDK

**Failures**

<details><summary>Example assertion failure</summary>
<pre>
shouldRedistributeDeploymentWhenDeploymentPartitionIsRestarted (Flaky Test): io.camunda.zeebe.it.clustering.DeploymentClusteredTest
java.lang.AssertionError: 

Expecting Optional to contain a value but it was empty.
	at io.camunda.zeebe.it.clustering.DeploymentClusteredTest.shouldRedistributeDeploymentWhenDeploymentPartitionIsRestarted(DeploymentClusteredTest.java:132)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
	at java.base/java.lang.Thread.run(Thread.java:1583)
</pre>
</details>

**Hypotheses**
I don't have an idea

**Logs**

<details><summary>Logs</summary>

[LOGS](https://drive.google.com/file/d/1ktAqb-OAA14cOd6Lo9z3sNFkf_cSfjvC/view?usp=drive_link)
</details>

**Comment by megglos** [comment]
ZPA-Triage:
- needs to be clarified in which domain this falls, we should investigate and follow-up with ZDP if this relates to the platform not the engine

**Comment by korthout** [comment]
@mustafadagher I'm not allowed to access the LOGS:
>Server failed to authenticate the request

Typically we upload the logs to our [Google Drive](https://drive.google.com/drive/folders/1_5JpW6NvCn16LgwclycXOwOUIwP2AC8a) and share a link to that.

**Comment by mustafadagher** [comment]
> @mustafadagher I'm not allowed to access the LOGS:
> 
> > Server failed to authenticate the request
> 
> Typically we upload the logs to our [Google Drive](https://drive.google.com/drive/folders/1_5JpW6NvCn16LgwclycXOwOUIwP2AC8a) and share a link to that.

Sorry for that! [here](https://drive.google.com/file/d/1ktAqb-OAA14cOd6Lo9z3sNFkf_cSfjvC/view?usp=drive_link) you go.

**Comment by oleschoenburg** [comment]
I've read the test code and tried to reproduce locally without success.
Since we don't have useful logs here and this could very well be a infrastructure related flake, I'll close the issue until it occurs again.

**Comment by korthout** [comment]
>Since we don't have useful logs here 

@oleschoenburg Do you know of any logs that might have helped investigate the issue?

**Comment by oleschoenburg** [comment]
Logs from the actual test execution. The linked logs are just the runner logs, mostly just showing the output of maven

**Comment by berkaycanbc** [comment]
Happened again:

CI logs: [link](https://github.com/camunda/camunda/actions/runs/9273314835/job/25513123286)
CI logs downloaded: [link](https://drive.google.com/drive/folders/1dfd-88sd2_yzyGjE5CM5IH3FkX8ESjPd)

**Comment by korthout** [comment]
>I have to reopen this as it happened again on main:

Closing again, as this run didn't include the fix yet:

- The run checked out 2f84c1b https://github.com/camunda/camunda/actions/runs/9503816673/job/26195095580
- The fix is 7fc6cb6
- 7fc6cb6 is not an ancestor of 2f84c1b

```
$ git merge-base 7fc6cb6 --is-ancestor 2f84c1b && echo yes || echo no
no
```
